src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -422,6 +422,14 @@ robocopy /S /NP %EXECUTION_DIR%int\%(Identity)\ %EXECUTION_DIR%native\
       <PostExecutionTestCommandLines Include="move $(UAP_Results_Path)$(XunitTestAssembly).xml .\$(XunitResultsFileName)" />
     </ItemGroup>
 
+    <!-- If the PreExecutionTestScript property is set, then it should be set to the full path to a script that will be directly incorporated
+         into the generated runtests.cmd script, immediately before the test is run. This can be used to set a number of JIT stress modes,
+         for example.
+     -->
+    <ItemGroup Condition="'$(PreExecutionTestScript)'!=''">
```

Comment:
Do keep in mind that this will copy all the contents into the runtest script whether it is on windows in to a cmd file or on linux into a sh script so the correct script will need to be given based on the OS.

Reply:
Also on linux the right thing might be to source the file in instead of copy it.

Reply:
On Windows, it appears as separate lines. E.g.:
```
:: ========================= BEGIN Test Execution =============================
echo Running tests... Start time: %TIME%
pushd %EXECUTION_DIR%
echo on
set COMPlus_AltJit=*
set COMPlus_AltJitNgen=*
set COMPlus_AltJitName=protononjit.dll
set COMPlus_NoGuiOnAssert=1
set COMPlus_ContinueOnAssert=0
set COMPlus_AltJitAssertOnNYI=1
set COMPlus_JitStress=1
call %RUNTIME_PATH%\dotnet.exe xunit.console.netcore.exe System.Composition.Tests.dll  -xml testResults.xml -notrait Benchmark=true -notrait category=nonnetcoreapptests -notrait category=nonwindowstests  -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing

@echo off
popd
echo Finished running tests.  End time=%TIME%, Exit code = %ERRORLEVEL%
EXIT /B %ERRORLEVEL%
:: ========================= END Test Execution =================================
```

> Also on linux the right thing might be to source the file in instead of copy it.

I think we'd want the same behavior as on Windows: that is, the runtests.cmd includes the contents of the file referenced at build time.

Yes, on Linux, you need to provide a file with the correct line endings.